test/integration/payments.js

```diff
@@ -0,0 +1,157 @@
+      const fixtures = require('../fixtures/' + k)
+      const { depends } = fixtures.dynamic
+      const fn = bitcoin.payments[k]
+
+      const base = {}
+      if (depends.pubkey) base.pubkey = keyPairs[0].publicKey
+      if (depends.pubkeys) base.pubkeys = keyPairs.map(x => x.publicKey)
+      if (depends.m) base.m = base.pubkeys.length
+      if (depends.data) base.data = [Buffer.from('bitcoinjs', 'utf8')]
+
+      const { output } = fn(base)
+      if (!output) throw new TypeError('Missing output')
+
+      const p2sh = bitcoin.payments.p2sh({ redeem: { output }, network: NETWORK })
+
+      regtestUtils.faucetComplex(p2sh.address, 5e4, (err, unspent) => {
+        if (err) return done(err)
+
+        const txb = new bitcoin.TransactionBuilder(NETWORK)
+        txb.addInput(unspent.txId, unspent.vout, null, p2sh.output) // NOTE: [prev]output
+        txb.addOutput(regtestUtils.RANDOM_ADDRESS, 2e4)
+
+        if (depends.signatures) {
+          keyPairs.forEach((keyPair) => {
+            txb.sign(0, keyPair, p2sh.redeem.output, null, unspent.value)
+          })
+        } else if (depends.signature) {
+          txb.sign(0, keyPairs[0], p2sh.redeem.output, null, unspent.value)
+        }
+
+        regtestUtils.broadcast(txb.build().toHex(), done)
+      })
+    })
+
+    it('can (wrapped in P2WSH) broadcast as an output, and be spent as an input', (done) => {
+      if (k === 'p2wpkh') return done() // skip P2WSH(P2WPKH)
+
+      const fixtures = require('../fixtures/' + k)
+      const { depends } = fixtures.dynamic
+      const fn = bitcoin.payments[k]
+
+      const base = {}
+      if (depends.pubkey) base.pubkey = keyPairs[0].publicKey
+      if (depends.pubkeys) base.pubkeys = keyPairs.map(x => x.publicKey)
+      if (depends.m) base.m = base.pubkeys.length
+      if (depends.data) base.data = [Buffer.from('bitcoinjs', 'utf8')]
+
+      const { output } = fn(base)
+      if (!output) throw new TypeError('Missing output')
+
+      const p2wsh = bitcoin.payments.p2wsh({ redeem: { output }, network: NETWORK })
+
+      regtestUtils.faucetComplex(p2wsh.address, 5e4, (err, unspent) => {
+        if (err) return done(err)
+
+        const txb = new bitcoin.TransactionBuilder(NETWORK)
+        txb.addInput(unspent.txId, unspent.vout, null, p2wsh.output) // NOTE: [prev]output
+        txb.addOutput(regtestUtils.RANDOM_ADDRESS, 2e4)
+
+        if (depends.signatures) {
+          keyPairs.forEach((keyPair) => {
+            txb.sign(0, keyPair, null, null, unspent.value, p2wsh.redeem.output)
+          })
+        } else if (depends.signature) {
+          txb.sign(0, keyPairs[0], null, null, unspent.value, p2wsh.redeem.output)
+        }
+
+        regtestUtils.broadcast(txb.build().toHex(), done)
+      })
+    })
+
+    it('can (wrapped in P2WSH, then wrapped in P2SH) broadcast as an output, and be spent as an input', (done) => {
+      if (k === 'p2wpkh') return done() // skip P2SH(P2WSH(P2WPKH))
```

Comment:
same here, is `P2SH(P2WSH(P2WPKH))` possible?
Are we simply lacking support? (it fails)